akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -339,17 +340,11 @@ private[remote] class Association(
         case OptionVal.Some(ref) => ref.cachedAssociation = null // don't use this Association instance any more
         case OptionVal.None      =>
       }
-      if (log.isDebugEnabled) {
-        val reason =
-          if (removed) "removed unused quarantined association"
-          else s"overflow of send queue, size [$qSize]"
-        log.debug(
-          "Dropping message [{}] from [{}] to [{}] due to {}",
-          Logging.messageClassName(message),
-          sender.getOrElse(deadletters),
-          recipient.getOrElse(recipient),
-          reason)
-      }
+      val reason =
+        if (removed) "Due to removed unused quarantined association"
+        else s"Due to overflow of send queue, size [$qSize]"
+      transport.system.eventStream.publish(Dropped(message, reason, recipient.getOrElse(deadletters)))
```

Comment:
This would supersede the idea of having a hi-frequency logger: https://github.com/akka/akka/pull/25594